Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -198,43 +219,54 @@ - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id <UIV
     } completion:NULL];
 }
 
-#pragma mark - UICollectionViewDelegate
+#pragma mark - Article Selection
 
-- (void)collectionView:(UICollectionView*)collectionView didSelectItemAtIndexPath:(NSIndexPath*)indexPath {
-    self.selectedArticle = [self.dataSource articleForIndexPath:indexPath];
+- (void)showArticle:(MWKArticle*)article discoveryMethod:(MWKHistoryDiscoveryMethod)discoveryMethod {
+    if (self.delegate) {
+        [self.delegate didSelectArticle:article sender:self];
+        return;
+    }
 
     WMFArticleContainerViewController* container =
```

Comment:
another implicit question: do we continue the trend of `WMFArticleContainerViewController` becoming the new home of `WebViewController` functionality, such as history list updates, logging, etc.?